c2rust-refactor/src/scripting/ast_visitor.rs

```diff
@@ -0,0 +1,655 @@
+            },
+        }
+
+        Ok(())
+    }
+
+    pub fn visit_expr(&self, expr: LuaTable<'lua>) -> LuaResult<()> {
+        call_lua_visitor_method!(self.visitor_obj,visit_expr(expr));
+
+        match expr.get::<_, String>("kind")?.as_str() {
+            "Box" => {
+                let boxed: LuaTable = expr.get("expr")?;
+
+                self.visit_expr(boxed)?;
+            },
+            "AssignOp"
+            | "Binary"
+            | "Assign" => {
+                let lhs: LuaTable = expr.get("lhs")?;
+                let rhs: LuaTable = expr.get("rhs")?;
+
+                self.visit_expr(lhs)?;
+                self.visit_expr(rhs)?;
+            },
+            "Array" => {
+                let values: LuaTable = expr.get("values")?;
+
+                for val in values.sequence_values::<LuaTable>() {
+                    self.visit_expr(val?)?;
+                }
+            },
+            "Path" => {
+                // TODO
+            },
+            "Lit" => {
+                // TODO: self.visit_literal(lit)?;
+            },
+            ref e => warn!("visit_expr: Found unsupported expr {}", e),
+        }
+
+        Ok(())
+    }
+
+    pub fn visit_stmt(&self, stmt: LuaTable<'lua>) -> LuaResult<()> {
+        call_lua_visitor_method!(self.visitor_obj,visit_stmt(stmt));
+
+        match stmt.get::<_, String>("kind")?.as_str() {
+            "Expr"
+            | "Semi" => {
+                let expr: LuaTable = stmt.get("expr")?;
+
+                self.visit_expr(expr)?;
+            },
+            "Local" => {
+                self.visit_local(stmt)?;
+            },
+            "Item" => {
+                let item: LuaTable = stmt.get("item")?;
+
+                self.visit_item(item)?;
+            },
+            ref e => warn!("visit_stmt: Unsupported Stmt kind: {}", e),
+        }
+
+        Ok(())
+    }
+
+    pub fn visit_local(&self, local: LuaTable<'lua>) -> LuaResult<()> {
+        call_lua_visitor_method!(self.visitor_obj,visit_local(local));
+
+        let opt_init: Option<LuaTable> = local.get("init")?;
+
+        if let Some(init) = opt_init {
+            self.visit_expr(init)?;
+        }
+
+        Ok(())
+    }
+
+    pub fn visit_block(&self, block: LuaTable<'lua>) -> LuaResult<()> {
+        call_lua_visitor_method!(self.visitor_obj,visit_block(block));
+
+        let stmts: LuaTable = block.get("stmts")?;
+
+        for stmt in stmts.sequence_values::<LuaTable>() {
+            self.visit_stmt(stmt?)?;
+        }
+
+        Ok(())
+    }
+
+    pub fn visit_fn_like(&self, fn_like: LuaTable<'lua>) -> LuaResult<()> {
+        call_lua_visitor_method!(self.visitor_obj,visit_fn_like(fn_like));
+
+        let block: LuaTable = fn_like.get("block")?;
+
+        self.visit_block(block)?;
+
+        Ok(())
+    }
+
+    pub fn finish(&self) -> LuaResult<()> {
+        call_lua_visitor_method!(self.visitor_obj,finish());
+
+        Ok(())
+    }
+}
+
+impl<'lua> IntoLuaAst<'lua> for FnLike {
```

Comment:
I think we should move all the `IntoLuaAst` impls into their own module so they're not spread between scripting.rs and ast_visitor.rs

Reply:
Good idea - do you think we should just have one `traits.rs` with scripting traits and impls? Or split them up into `traits/into_lua_ast.rs` and friends?

Reply:
Just call it `scripting/into_lua_ast.rs`